picard.go

```diff
@@ -196,8 +196,9 @@ func (p PersistenceORM) upsert(data interface{}, deleteFilters interface{}) erro
 	if deleteFilters != nil {
 		deletes := []dbchange.Change{}
 		deleteResults, err := p.FilterModel(FilterRequest{
-			FilterModel: deleteFilters,
-			Runner:      p.transaction,
+			FilterModel:  deleteFilters,
+			Runner:       p.transaction,
+			Associations: getAssociations(tableMetadata),
```

Comment:
If you look further down this code at line 225/226 in this file, you'll see that we find an object key for a record based on these deleteResults. Many of the Structured REST items have deeply nested lookups, so we need to add that nested information on this query so that the object key can be properly created.